src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -2199,6 +2427,24 @@ private static void addLastSibling(DetailAstImpl self, DetailAstImpl sibling) {
         }
     }
 
+    /**
+     * Builds an imaginary DetailAstImpl with the given token details.
+     *
+     * @param tokenType the token type of this DetailAstImpl
+     * @param text the text of this DetailAstImpl
+     * @param lineNumber the line number of this DetailAstImpl
+     * @param columnNumber the column number of this DetailAstImpl
+     * @return imaginary DetailAstImpl from given details
+     */
+    private static DetailAstImpl buildImaginaryWithDetails(
```

Comment:
Going on the previous comment, why not just name all these `createImaginary` with different parameters to fill. It would make this almost like a builder method.

Reply:
Yep, done